core/src/main/java/com/opensymphony/xwork2/config/impl/AbstractMatcher.java

```diff
@@ -192,7 +217,7 @@ protected String convertParam(String val, Map<String, String> vars) {
             c = val.charAt(x);
             if (x < len - 2 && 
                     c == '{' && '}' == val.charAt(x+2)) {
-                varVal = (String)vars.get(String.valueOf(val.charAt(x + 1)));
+                varVal = vars.get(String.valueOf(val.charAt(x + 1)));
```

Comment:
(NON-BLOCKER) Never thought about it or ran in to it, but do we throw a startup message or error if it's something *other* than 0-9 or has a spurious `{42}` or `{42` or is that just considered part of a valid, if silly, path?

Reply:
Hm... good point, I think we can extract `OgnlTextParser` logic and reuse it here

Reply:
https://issues.apache.org/jira/browse/WW-5073